app/models/reports/facility_state.rb

```diff
@@ -1,6 +1,12 @@
 module Reports
   class FacilityState < Matview
     self.table_name = "reporting_facility_states"
+
     belongs_to :facility
+
+    def self.for_region(region_or_source)
+      region = region_or_source.region
+      where("#{region.region_type}_region_id" => region.id)
+    end
```

Comment:
Is it worth unit testing this? I would find those tests as valuable documentation for sure.

Reply:
I'd love to have these tested with some straightforward specs, for sure.